packages/world/src/systems/MineSystem.sol

```diff
@@ -74,7 +74,9 @@ contract MineSystem is System {
       // TODO: factor out the mass reduction logic so it's cheaper to call
       EntityId entityId = _mine(caller, coord, toolSlot, extraData);
       massLeft = Mass._getMass(entityId);
-    } while (massLeft > 0 && Energy._getEnergy(caller) > 0);
+    } while (
+      massLeft > 0 && Energy._getEnergy(caller) > 0 && InventoryUtils.getToolData(caller, toolSlot).massLeft > 0
```

Comment:
will this work if mining with hand? massLeft will be 0

Reply:
ah I guess it will just mine once

Reply:
oh wait true, with hand it should keep mining lol

Reply:
actually this is only in the method that passes in a tool slot, we dont pass one in to the hand case